lib/http.js

```diff
@@ -73,12 +75,16 @@ HttpClient.prototype.buildRequest = function(rurl, data, exheaders, exoptions) {
 
   exoptions = exoptions || {};
   for (attr in exoptions) {
-    if (mergeOptions.indexOf(attr) !== -1) {
-      for (header in exoptions[attr]) {
-        options[attr][header] = exoptions[attr][header];
+    if (exoptions.hasOwnProperty(attr)) {
+      if (mergeOptions.indexOf(attr) !== -1) {
```

Comment:
Could you please combine both checks as we don't have an `else` for the second condition?
(`if (exoptions.hasOwnProperty(attr) && mergeOptions.indexOf(attr) !== -1)`)